gnocchi/incoming/redis.py

```diff
@@ -108,3 +108,15 @@ def process_measure_for_metric(self, metric):
 
         # ltrim is inclusive, bump 1 to remove up to and including nth item
         self._client.ltrim(key, item_len + 1, -1)
+
+    def iter_on_sacks_to_process(self):
+        self._client.config_set("notify-keyspace-events", "El")
+        p = self._client.pubsub()
+        db = self._client.connection_pool.connection_kwargs['db']
+        channel = b"__keyevent@" + str(db).encode() + b"__:rpush"
+        p.subscribe(channel)
+        for message in p.listen():
+            if message['type'] == 'message' and message['channel'] == channel:
```

Comment:
we're only subscribed to a single channel so we should only get that channel no?

Reply:
Yes though you never know if e.g. Redis starts publishing other stuff by default or whatever. Better be safe then sorry :)